pkg/code-generator/codegen/templates/snapshot_emitter_template.go

```diff
@@ -275,7 +278,7 @@ func (c *{{ lower_camel .GoName }}Emitter) Snapshots(watchNamespaces []string, o
 			if err != nil {
 				contextutils.LoggerFrom(ctx).Panicw("error while hashing, this should never happen", zap.Error(err))
 			}
-			if previousHash == currentHash {
+			if !needsSync && previousHash == currentHash {
```

Comment:
When our current and previous hash are the same why would we need to sync? can we add a comment on that

Reply:
Nevermind I see its due to the forceemit change